### PR TITLE
feature/STJS-622-standalone-cybertonica-implementation

### PR DESCRIPTION
**Pull request by pgs-gmartowski**

### CODE REVIEW

File: src/application/core/integrations/Cybertonica.ts [hunk 1]
```diff
@@ -6,6 +6,9 @@ import { environment } from '../../../environments/environment';
 import { BrowserLocalStorage } from '../../../shared/services/storage/BrowserLocalStorage';
 import { Language } from '../shared/Language';
 import { ICybertonica } from './ICybertonica';
+import { interval, Observable } from 'rxjs';
```
Comment (wirus15) [inline src/application/core/integrations/Cybertonica.ts:9]: are those imports used?

File: src/application/core/shared/MessageBus.ts [hunk 2]
```diff
@@ -42,27 +42,28 @@ export class MessageBus implements Subscribable<IMessageBusEvent> {
...
     BY_PASS_CARDINAL: 'BY_PASS_CARDINAL',
     BY_PASS_INIT: 'BY_PASS_INIT',
+    CALL_MERCHANT_ERROR_CALLBACK: 'CALL_MERCHANT_ERROR_CALLBACK',
+    CALL_MERCHANT_CANCEL_CALLBACK: 'CALL_MERCHANT_CANCEL_CALLBACK',
+    CALL_MERCHANT_SUCCESS_CALLBACK: 'CALL_MERCHANT_SUCCESS_CALLBACK',
+    CALL_MERCHANT_SUBMIT_CALLBACK: 'CALL_MERCHANT_SUBMIT_CALLBACK',
     CALL_SUBMIT_EVENT: 'CALL_SUBMIT_EVENT',
+    CONFIG_CHECK: 'ST_CONFIG_CHECK',
     DESTROY: 'DESTROY',
+    GET_TID: 'GET_TID',
```
Comment (wirus15) [inline src/application/core/shared/MessageBus.ts:52]: I would rename this to `GET_CYBERTONICA_TID` to make it more explicit 😉

File: src/client/ST.ts [hunk 3]
```diff
@@ -158,6 +162,19 @@ class ST {
...
   }
 
+  public Cybertonica(): Promise<string> {
+    return new Promise(resolve =>
+      this._framesHub
+        .waitForFrame(Selectors.CONTROL_FRAME_IFRAME)
+        .pipe(
+          switchMap((controlFrame: string) =>
+            from(this._communicator.query({ type: MessageBus.EVENTS_PUBLIC.GET_TID }, controlFrame))
+          )
```
Comment (wirus15) [inline src/client/ST.ts:172]: I think you should add the `first()` operator here to make sure the observable completes after the value is resolved

File: src/application/core/integrations/Cybertonica.ts (hunk 1 above)
Comment (pgs-gmartowski) [reply src/application/core/integrations/Cybertonica.ts:9]: nope :) deleted

File: src/application/core/shared/MessageBus.ts (hunk 2 above)
Comment (pgs-gmartowski) [reply src/application/core/shared/MessageBus.ts:52]: true, done

File: src/client/ST.ts (hunk 3 above)
Comment (pgs-gmartowski) [reply src/client/ST.ts:172]: I think I fixed it :)

Comment (wirus15) [reply src/client/ST.ts:172]: I'm looking at this code now and I'm wondering why I thought `first()` is needed here 🤔 It's an observable created from promise so it completes right after it resolves 🤷‍♂️ Anyway, this doesn't change anything so it's up to you whether you want to remove it 😁

**Pull request merged**